wagtail/documents/views/bulk_actions/add_to_collection.py

```diff
@@ -39,8 +39,8 @@ def get_execution_context(self):
         }
 
     @classmethod
-    def execute_action(cls, objects, **kwargs):
-        cls.collection = kwargs.get('collection', None)
+    def execute_action(cls, objects, collection=None):
+        cls.collection = collection
```

Comment:
There seems to still be a lot of storing variables on the class here. Is that necessary? If we can, I think we should just use local variables here (and in the other bulk actions that use class attributes)

Reply:
I had a similar thought about this specific one. The collection name is used in the success message. Two options:
1. make the success message generic
2. come up with a different way of passing context

Reply:
Personally, I'd favour 2 - since `collection` is being passed in, it seems reasonable to set eg `self.collection` elsewhere, and use that in the success message (or give that the `execution_context` as kwargs as well, I suppose, but doing it on the instance is probably simpler). As for the others that store the number of updated objects on the class, that seems like a reasonable variable for `execute_action` to return. What do you think?

Reply:
> As for the others that store the number of updated objects on the class, that seems like a reasonable variable for execute_action to return

Are you referring to the `num_parent_objects` and `num_child_objects` variables? If so, then yes I think returning them as output for the `execute_action` method makes sense

Reply:
Yes, that's what I was thinking of.